In [1]:
import pandas as pd
pd.set_option('display.max_columns',None,'display.max_rows',None)

In [2]:
data = pd.read_csv("C:\\Users\\gaber\\Downloads\\Uncleaned_DS_jobs\\Uncleaned_DS_jobs.csv")
data.shape

(672, 15)

In [3]:
data.dtypes

index                  int64
Job Title             object
Salary Estimate       object
Job Description       object
Rating               float64
Company Name          object
Location              object
Headquarters          object
Size                  object
Founded                int64
Type of ownership     object
Industry              object
Sector                object
Revenue               object
Competitors           object
dtype: object

In [4]:
data.isna().sum()

index                0
Job Title            0
Salary Estimate      0
Job Description      0
Rating               0
Company Name         0
Location             0
Headquarters         0
Size                 0
Founded              0
Type of ownership    0
Industry             0
Sector               0
Revenue              0
Competitors          0
dtype: int64

In [5]:
data.head(10)

index                                 Job Title  \
0      0                         Sr Data Scientist   
1      1                            Data Scientist   
2      2                            Data Scientist   
3      3                            Data Scientist   
4      4                            Data Scientist   
5      5                            Data Scientist   
6      6  Data Scientist / Machine Learning Expert   
7      7                            Data Scientist   
8      8          Staff Data Scientist - Analytics   
9      9                            Data Scientist   

                Salary Estimate  \
0  $137K-$171K (Glassdoor est.)   
1  $137K-$171K (Glassdoor est.)   
2  $137K-$171K (Glassdoor est.)   
3  $137K-$171K (Glassdoor est.)   
4  $137K-$171K (Glassdoor est.)   
5  $137K-$171K (Glassdoor est.)   
6  $137K-$171K (Glassdoor est.)   
7  $137K-$171K (Glassdoor est.)   
8  $137K-$171K (Glassdoor est.)   
9  $137K-$171K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Description\n\nThe Senior Data Scientist is re...     3.1   
1  Secure our Nation, Ignite your Future\n\nJoin ...     4.2   
2  Overview\n\n\nAnalysis Group is one of the lar...     3.8   
3  JOB DESCRIPTION:\n\nDo you have a passion for ...     3.5   
4  Data Scientist\nAffinity Solutions / Marketing...     2.9   
5  About Us:\n\nHeadquartered in beautiful Santa ...     4.2   
6  Posting Title\nData Scientist / Machine Learni...     3.9   
7  Introduction\n\nHave you always wanted to run ...     3.5   
8  Intuit is seeking a Staff Data Scientist to co...     4.4   
9  Ready to write the best chapter of your career...     3.6   

              Company Name           Location            Headquarters  \
0         Healthfirst\n3.1       New York, NY            New York, NY   
1             ManTech\n4.2      Chantilly, VA             Herndon, VA   
2      Analysis Group\n3.8         Boston, MA              Boston, MA   
3             INFICON\n3.5         Newton, MA  Bad Ragaz, Switzerland   
4  Affinity Solutions\n2.9       New York, NY            New York, NY   
5         HG Insights\n4.2  Santa Barbara, CA       Santa Barbara, CA   
6            Novartis\n3.9      Cambridge, MA      Basel, Switzerland   
7              iRobot\n3.5        Bedford, MA             Bedford, MA   
8       Intuit - Data\n4.4      San Diego, CA       Mountain View, CA   
9  XSELL Technologies\n3.6        Chicago, IL             Chicago, IL   

                      Size  Founded        Type of ownership  \
0   1001 to 5000 employees     1993   Nonprofit Organization   
1  5001 to 10000 employees     1968         Company - Public   
2   1001 to 5000 employees     1981  Private Practice / Firm   
3    501 to 1000 employees     2000         Company - Public   
4      51 to 200 employees     1998        Company - Private   
5      51 to 200 employees     2010        Company - Private   
6         10000+ employees     1996         Company - Public   
7   1001 to 5000 employees     1990         Company - Public   
8  5001 to 10000 employees     1983         Company - Public   
9      51 to 200 employees     2014        Company - Private   

                                   Industry                     Sector  \
0                        Insurance Carriers                  Insurance   
1                    Research & Development          Business Services   
2                                Consulting          Business Services   
3     Electrical & Electronic Manufacturing              Manufacturing   
4                   Advertising & Marketing          Business Services   
5              Computer Hardware & Software     Information Technology   
6                 Biotech & Pharmaceuticals  Biotech & Pharmaceuticals   
7  Consumer Electronics & Appliances Stores                     Retail   
8              Computer Hardware & Software     Information Technology   
9   Enterprise Software & Network Solutions     Information Technolo

In [6]:
data.columns

Index(['index', 'Job Title', 'Salary Estimate', 'Job Description', 'Rating',
       'Company Name', 'Location', 'Headquarters', 'Size', 'Founded',
       'Type of ownership', 'Industry', 'Sector', 'Revenue', 'Competitors'],
      dtype='object')

## Remove the index column

In [7]:
data.drop(['index'], axis= 1, inplace= True)
data.shape

(672, 14)

## Splitting the Salary Estimate to 2 columns
#### -We are splitting the column to Min Salar and Max Salary
#### -Creating a Mean_salary column

In [8]:
data['Salary Estimate'] = data['Salary Estimate'].str.replace(' (Glassdoor est.)','')
data['Salary Estimate'].head()

0    $137K-$171K
1    $137K-$171K
2    $137K-$171K
3    $137K-$171K
4    $137K-$171K
Name: Salary Estimate, dtype: object

In [9]:
data[['Min_Salary', 'Max_Salary']] = data['Salary Estimate'].str.split('-', expand=True)
data[['Min_Salary', 'Max_Salary']].head()

Min_Salary Max_Salary
0      $137K      $171K
1      $137K      $171K
2      $137K      $171K
3      $137K      $171K
4      $137K      $171K

#### As we can see from the cell below that there are other string values in the Max_Salary column that need to be removed before trimming and changing the dtype

In [10]:
print(data['Min_Salary'].unique())
print(data['Max_Salary'].unique())

['$137K' '$75K' '$79K' '$99K' '$90K' '$101K' '$56K' '$71K' '$91K' '$141K'
 '$145K' '$122K' '$112K' '$110K' '$124K' '$69K' '$31K' '$95K' '$212K'
 '$66K' '$128K' '$138K' '$80K' '$87K' '$92K' '$105K']
['$171K' '$131K' '$132K' '$109K' '$165K' '$97K' '$106K' '$123K' '$124K'
 '$150K' '$225K' '$225K(Employer est.)' '$147K' '$146K' '$116K' '$163K'
 '$198K' '$133K' '$56K' '$119K' '$331K' '$112K' '$201K' '$158K' '$141K'
 '$155K' '$167K']


In [11]:
data['Max_Salary'] = data['Max_Salary'].str.replace('(Employer est.)','')

In [12]:
data[['Min_Salary', 'Max_Salary']] = data[['Min_Salary', 'Max_Salary']].apply(lambda x: x.str.strip('$K')) 
data[['Min_Salary', 'Max_Salary']] = data[['Min_Salary', 'Max_Salary']].astype(int)
data[['Min_Salary', 'Max_Salary']] = data[['Min_Salary', 'Max_Salary']] * 1000
print(data['Min_Salary'].dtype, data['Max_Salary'].dtype)

int32 int32


In [13]:
data[['Min_Salary', 'Max_Salary']].head()

Min_Salary  Max_Salary
0      137000      171000
1      137000      171000
2      137000      171000
3      137000      171000
4      137000      171000

In [14]:
data['Average_Salary'] = (data['Min_Salary'] + data['Max_Salary']) / 2

In [15]:
data.drop(columns= ['Salary Estimate'], axis=1, inplace= True)

##  Size of employees:
#### - change to Mean of employees
#### - Change the dtype to int 

In [16]:
data['Size'] = data['Size'].str.replace(' employees','')

In [17]:
data['Size'].unique()

array(['1001 to 5000', '5001 to 10000', '501 to 1000', '51 to 200',
       '10000+', '201 to 500', '1 to 50', '-1', 'Unknown'], dtype=object)

In [18]:
data_map = {'1001 to 5000': 2500, '5001 to 10000': 5000, '501 to 1000': 750, '51 to 200': 125,
       '10000+': 10000, '201 to 500': 350, '1 to 50': 25, '-1': 0, 'Unknown': 0}
data['Size'] = data['Size'].map(data_map)
data.rename(columns={'Size':'Average_size_of_employees'}, inplace= True)
data['Average_size_of_employees'].head()

0    2500
1    5000
2    2500
3     750
4     125
Name: Average_size_of_employees, dtype: int64

#### We have some companies with 0 number of employees which does not make sense

In [19]:
data['Average_size_of_employees'].value_counts()

Average_size_of_employees
125      135
2500     104
25        86
350       85
10000     80
750       77
5000      61
0         44
Name: count, dtype: int64

#### List of companies with no employee range 

In [20]:
data[data['Average_size_of_employees'] == 0]['Company Name']

154             Covid-19 Search Partners
158                  Radical Convergence
189                Constant Contact\n3.6
193                      SkillSoniq\n5.0
258           The Drive Media, Inc.\n5.0
261                Constant Contact\n3.6
274                      SkillSoniq\n5.0
282                       Trovetechs Inc
285                       Trovetechs Inc
308                  Kollasoft Inc.\n3.2
351                     Point72 Ventures
357                       Hatch Data Inc
358                       Hatch Data Inc
359                       Hatch Data Inc
360                       Hatch Data Inc
361                       Hatch Data Inc
362                       Hatch Data Inc
388                               HireAi
389                               HireAi
409                                Navio
424                           Voice\n3.4
430       Falcon IT & Staffing Solutions
444                       Hive (CA)\n2.1
459                       Blue Cloak LLC
495             

#### As we can see there are many company names are redundunt so we need to drop the duplicate companies and remove the companies with no names 

In [21]:
print(data.shape)
data.drop_duplicates(subset=['Company Name'], keep= 'first', inplace=True)
print(data.shape)
data = data.drop(data[(data['Company Name'] == '') | (data['Company Name'] == ' ')].index)
print(data.shape)

(672, 16)
(432, 16)
(432, 16)


### The companies with no number of employees might be classified data or startups. Whether we impute or drop those companies rely on the domain knowledge and the specific analysis we might do later.
___

## Revnue Column
#### The Revenue column has to a number so we can do analysis on different comapnies in the future. We changed every value in the Revenue column to represent roughly the average value of the companies revenue.

In [22]:
data['Revenue'].unique()

array(['Unknown / Non-Applicable', '$1 to $2 billion (USD)',
       '$100 to $500 million (USD)', '$10+ billion (USD)',
       '$2 to $5 billion (USD)', '$500 million to $1 billion (USD)',
       '$5 to $10 billion (USD)', '$10 to $25 million (USD)',
       '$25 to $50 million (USD)', '$50 to $100 million (USD)',
       '$1 to $5 million (USD)', '$5 to $10 million (USD)',
       'Less than $1 million (USD)', '-1'], dtype=object)

In [23]:
revenue_map = {'Unknown / Non-Applicable':0, '$1 to $2 billion (USD)':1.5*(10**3),
       '$100 to $500 million (USD)': 300, '$10+ billion (USD)': 10**4,
       '$2 to $5 billion (USD)': 3.5*(10**3), '$500 million to $1 billion (USD)': 750,
       '$5 to $10 billion (USD)': 7.5*(10**3), '$10 to $25 million (USD)': 17.5,
       '$25 to $50 million (USD)': 37.5, '$50 to $100 million (USD)': 75,
       '$1 to $5 million (USD)': 3, '$5 to $10 million (USD)': 7.5,
       'Less than $1 million (USD)': 1, '-1':0}
data['Revenue_in_Million'] = data['Revenue'].map(revenue_map)
data['Revenue_in_Million'].dtype

dtype('float64')

In [24]:
data.drop(columns=['Revenue'], axis= 1, inplace= True)

### In the cell below there are 155 companies with 0 revenue which indicates that it is either classified information or startup companies. One way to do this is to impute those missing values using the mean of the revenue.
### You might also leave it as is if it is not that important in your analysis.

In [25]:
data['Revenue_in_Million'].value_counts()

Revenue_in_Million
0.0        157
300.0       60
10000.0     34
17.5        31
37.5        25
1500.0      23
75.0        22
3500.0      20
3.0         19
7.5         12
1.0         11
750.0       10
7500.0       8
Name: count, dtype: int64

In [26]:
mask = data['Revenue_in_Million'] == 0
data.loc[mask, 'Revenue_in_Million'] = data['Revenue_in_Million'].mean()

In [27]:
data['Revenue_in_Million'].value_counts()

Revenue_in_Million
1234.462963     157
300.000000       60
10000.000000     34
17.500000        31
37.500000        25
1500.000000      23
75.000000        22
3500.000000      20
3.000000         19
7.500000         12
1.000000         11
750.000000       10
7500.000000       8
Name: count, dtype: int64

## Remove the numbers from the company name

In [28]:
data['Company Name'].unique()

array(['Healthfirst\n3.1', 'ManTech\n4.2', 'Analysis Group\n3.8',
       'INFICON\n3.5', 'Affinity Solutions\n2.9', 'HG Insights\n4.2',
       'Novartis\n3.9', 'iRobot\n3.5', 'Intuit - Data\n4.4',
       'XSELL Technologies\n3.6', 'Novetta\n4.5', '1904labs\n4.7',
       'PNNL\n3.7', 'Old World Industries\n3.1',
       'Mathematica Policy Research\n3.4',
       'Guzman & Griffin Technologies (GGTI)\n4.4',
       'Upside Business Travel\n4.1', 'Buckman\n3.5',
       'Insight Enterprises, Inc.\n4.2', 'Tower Health\n3.5',
       'Triplebyte\n3.2', 'PulsePoint\n4.3', 'Exponent\n3.5',
       'Guardian Life\n3.5',
       'Spectrum Communications and Consulting\n3.4',
       'Oversight Systems\n4.7', 'LSQ\n4.2',
       'MIT Lincoln Laboratory\n3.8', 'Kingfisher Systems\n4.5',
       'Formation\n2.8', 'Cohere Health\n5.0', 'Acuity Insurance\n4.8',
       'Chef\n3.6', 'Puget Sound Energy\n3.3', 'Sandhills Global\n2.7',
       'A Place for Mom\n2.7', 'Great-Circle Technologies\n2.2',
       'Edmu

In [29]:
data['Company Name'].head()

0           Healthfirst\n3.1
1               ManTech\n4.2
2        Analysis Group\n3.8
3               INFICON\n3.5
4    Affinity Solutions\n2.9
Name: Company Name, dtype: object

In [30]:
data['Company Name'] = data['Company Name'].apply(lambda x: x.split('\n')[0])
data['Company Name'].head()

0           Healthfirst
1               ManTech
2        Analysis Group
3               INFICON
4    Affinity Solutions
Name: Company Name, dtype: object

## Exracting important keywords from the Description column
#### The Description columns contains information about the job responsibilities and the technologies used in the applied position.

In [31]:
print(data['Job Description'][0])
print(data['Job Description'][1])
print(data['Job Description'][2])

Description

The Senior Data Scientist is responsible for defining, building, and improving statistical models to improve business processes and outcomes in one or more healthcare domains such as Clinical, Enrollment, Claims, and Finance. As part of the broader analytics team, Data Scientist will gather and analyze data to solve and address complex business problems and evaluate scenarios to make predictions on future outcomes and work with the business to communicate and support decision-making. This position requires strong analytical skills and experience in analytic methods including multivariate regressions, hierarchical linear models, regression trees, clustering methods and other complex statistical techniques.

Duties & Responsibilities:

• Develops advanced statistical models to predict, quantify or forecast various operational and performance metrics in multiple healthcare domains
• Investigates, recommends, and initiates acquisition of new data resources from internal and ex

In [32]:
data['Python'] = data['Job Description'].apply(lambda x: 1 if 'python' in x.lower() else 0)
data['Excel'] = data['Job Description'].apply(lambda x: 1 if 'excel' in x.lower() else 0)
data['AWS'] = data['Job Description'].apply(lambda x: 1 if 'aws' in x.lower() else 0)
data['Tableau'] = data['Job Description'].apply(lambda x: 1 if 'tableau' in x.lower() else 0)
data['SQL'] = data['Job Description'].apply(lambda x: 1 if 'sql' in x.lower() else 0)
data['Hadoop'] = data['Job Description'].apply(lambda x: 1 if 'hadoop' in x.lower() else 0)

In [33]:
data.head()

Job Title                                    Job Description  \
0  Sr Data Scientist  Description\n\nThe Senior Data Scientist is re...   
1     Data Scientist  Secure our Nation, Ignite your Future\n\nJoin ...   
2     Data Scientist  Overview\n\n\nAnalysis Group is one of the lar...   
3     Data Scientist  JOB DESCRIPTION:\n\nDo you have a passion for ...   
4     Data Scientist  Data Scientist\nAffinity Solutions / Marketing...   

   Rating        Company Name       Location            Headquarters  \
0     3.1         Healthfirst   New York, NY            New York, NY   
1     4.2             ManTech  Chantilly, VA             Herndon, VA   
2     3.8      Analysis Group     Boston, MA              Boston, MA   
3     3.5             INFICON     Newton, MA  Bad Ragaz, Switzerland   
4     2.9  Affinity Solutions   New York, NY            New York, NY   

   Average_size_of_employees  Founded        Type of ownership  \
0                       2500     1993   Nonprofit Organization   
1                       5000     1968         Company - Public   
2                       2500     1981  Private Practice / Firm   
3                        750     2000         Company - Public   
4                        125     1998        Company - Private   

                                Industry             Sector  \
0                     Insurance Carriers          Insurance   
1                 Research & Development  Business Services   
2                             Consulting  Business Services   
3  Electrical & Electronic Manufacturing      Manufacturing   
4                Advertising & Marketing  Business Services   

                                         Competitors  Min_Salary  Max_Salary  \
0            EmblemHealth, UnitedHealth Group, Aetna      137000      171000   
1                                                 -1      137000      171000   
2                                                 -1      137000      171000   
3  MKS Instruments, Pfeiffer Vacuum, Agilent Tech...      137000      171000   
4               Commerce Signals, Cardlytics, Yodlee      137000      171000   

   Average_Salary  Revenue_in_Million  Python  Excel  AWS  Tableau  SQL  \
0        154000.0         1234.462963       0      0    1        0    0   
1        154000.0         1500.000000       0      0    0        0    1   
2        154000.0          300.000000       1      1    1        0    0   
3        154000.0          300.000000       1      1    1        0    1   
4        154000.0         1234.462963       1      1    0        0    1   

   Hadoop  
0       0  
1       1  
2       0  
3       0  
4       0

## Age of the company from the founded column by subtracting it from the current year
#### Notice that some company's has -1 in the Founded column those we will set them to the current year

In [34]:
data['Founded'].value_counts()[0:5]

Founded
-1       79
 2012    17
 1999    14
 2000    13
 2010    13
Name: count, dtype: int64

In [35]:
current_year = 2023
data['Founded'] = data['Founded'].apply(lambda x: current_year if x == -1 else x)
data['Company_age'] = current_year - data['Founded']
data['Company_age'].head()

0    30
1    55
2    42
3    23
4    25
Name: Company_age, dtype: int64

## Headquarters and Location columns splitting the city from the state/province
#### We can see from the cell below that the Location column contains United States, Remote, New Jersey, Utah, and Texas as values.

In [36]:
data['Location'].value_counts()

Location
San Francisco, CA             36
New York, NY                  33
Washington, DC                18
Chicago, IL                   13
Boston, MA                    10
United States                  9
Cambridge, MA                  9
Chantilly, VA                  8
Reston, VA                     7
McLean, VA                     7
Herndon, VA                    6
Atlanta, GA                    5
Santa Clara, CA                5
San Diego, CA                  5
Saint Louis, MO                5
Palo Alto, CA                  4
Fort Belvoir, VA               4
Seattle, WA                    4
Plano, TX                      4
Remote                         3
Pittsburgh, PA                 3
San Carlos, CA                 3
Philadelphia, PA               3
Bellevue, WA                   3
Raleigh, NC                    3
South San Francisco, CA        3
Thousand Oaks, CA              3
Cincinnati, OH                 3
Omaha, NE                      2
Birmingham, AL                 2
S

In [37]:
data['Job_state'] = data['Location'].apply(lambda x: x.split(',')[-1])
data['Job_state'].head()

0     NY
1     VA
2     MA
3     MA
4     NY
Name: Job_state, dtype: object

In [38]:
data['Job_state'].value_counts()

Job_state
 CA              101
 VA               55
 NY               35
 MA               32
 MD               22
 IL               21
 DC               18
 TX               13
 WA               12
 PA               10
 MO                9
 OH                9
United States      9
 NJ                9
 NC                7
 GA                7
 CO                6
 WI                5
 IN                4
 FL                4
 TN                4
 AZ                4
 CT                3
 MI                3
 AL                3
 NE                3
Remote             3
 NH                2
 UT                2
 OR                2
 SC                2
 OK                2
 MN                2
Texas              1
 IA                1
 LA                1
 RI                1
 MS                1
New Jersey         1
Utah               1
 KS                1
California         1
Name: count, dtype: int64

In [39]:
data[(data['Location'] == 'United States') | (data['Location'] == 'Remote') | (data['Location'] == 'New Jersey')\
     | (data['Location'] == 'Utah') | (data['Location'] == 'Texas') | (data['Location'] == 'California')][['Location', 'Headquarters']]

Location           Headquarters
16          Remote         Washington, DC
34   United States            Seattle, WA
53          Remote             Vienna, VA
89   United States            Mc Lean, VA
93            Utah       South Jordan, UT
100     New Jersey            Holmdel, NJ
227  United States  Tel Aviv-Yafo, Israel
237          Texas           San Jose, CA
305         Remote      San Francisco, CA
307  United States         Scottsdale, AZ
438  United States            Livonia, MI
473  United States           Stamford, CT
483  United States             Reston, VA
510  United States        Los Angeles, CA
601     California        Los Angeles, CA
620  United States          San Diego, CA

In [40]:
def work_location_cleaning(title):
    if 'United States' == title:
        title = 'US'
    elif 'New Jersey' == title:
        title = 'NJ'
    elif 'Texas' == title:
        title = 'TX'
    elif 'Utah' == title:
        title = 'UT'
    elif 'California' == title:
        title = 'CA'
    return title
    
data['Job_state'] = data['Job_state'].map(work_location_cleaning) 
data['Job_state'].unique()

array([' NY', ' VA', ' MA', ' CA', ' IL', ' MO', ' WA', ' DC', 'Remote',
       ' TN', ' TX', ' PA', ' AZ', ' WI', ' GA', ' FL', 'US', ' NE',
       ' KS', ' NH', ' NJ', ' LA', ' OH', ' IN', ' MD', ' CO', 'UT', 'NJ',
       ' UT', ' OR', ' MI', ' SC', ' MS', ' AL', ' RI', ' IA', ' MN',
       ' OK', ' CT', ' NC', 'TX', 'CA'], dtype=object)

## Rating column

In [41]:
data['Rating'].value_counts().head()

Rating
 3.5    35
-1.0    30
 3.7    26
 5.0    24
 3.4    24
Name: count, dtype: int64

In [42]:
data['Rating'] = data['Rating'].map(lambda x: 0 if x == -1 else x)
data['Rating'].value_counts().head()

Rating
3.5    35
0.0    30
3.7    26
5.0    24
3.4    24
Name: count, dtype: int64

## Now for the -1 values in the headquarter.
### We will fill those values with the value in the location column

In [43]:
data['Headquarters'].value_counts().head()

Headquarters
New York, NY         25
San Francisco, CA    20
-1                   16
Reston, VA           10
Chicago, IL           9
Name: count, dtype: int64

In [44]:
mask = data['Headquarters'] == '-1'
location_df = data.loc[mask, 'Location']
data.loc[mask, 'Headquarters'] = location_df
data['Headquarters'].value_counts().head()

Headquarters
New York, NY         26
San Francisco, CA    23
Reston, VA           11
Chicago, IL           9
Boston, MA            8
Name: count, dtype: int64

## End of the cleaning process and saving the file
#### We removed the compitetors column from the data as it has too many -1 values more than half 

In [45]:
data['Competitors'].value_counts()

Competitors
-1                                                                                              324
Oak Ridge National Laboratory, National Renewable Energy Lab, Los Alamos National Laboratory      2
Intertek, SGS, Bureau Veritas                                                                     1
Kforce, PageGroup, Robert Half                                                                    1
MIT Lincoln Laboratory, Lockheed Martin, Northrop Grumman                                         1
Raytheon Technologies, Northrop Grumman, Booz Allen Hamilton                                      1
Seagate Technology, Toshiba                                                                       1
TASC, Vencore, Booz Allen Hamilton                                                                1
ACRT Services, Bartlett Tree Experts                                                              1
Archibus, iOffice, Planon                                                               

In [46]:
data.drop(columns=['Competitors'], axis=1 ,inplace= True)

In [47]:
data.head()

Job Title                                    Job Description  \
0  Sr Data Scientist  Description\n\nThe Senior Data Scientist is re...   
1     Data Scientist  Secure our Nation, Ignite your Future\n\nJoin ...   
2     Data Scientist  Overview\n\n\nAnalysis Group is one of the lar...   
3     Data Scientist  JOB DESCRIPTION:\n\nDo you have a passion for ...   
4     Data Scientist  Data Scientist\nAffinity Solutions / Marketing...   

   Rating        Company Name       Location            Headquarters  \
0     3.1         Healthfirst   New York, NY            New York, NY   
1     4.2             ManTech  Chantilly, VA             Herndon, VA   
2     3.8      Analysis Group     Boston, MA              Boston, MA   
3     3.5             INFICON     Newton, MA  Bad Ragaz, Switzerland   
4     2.9  Affinity Solutions   New York, NY            New York, NY   

   Average_size_of_employees  Founded        Type of ownership  \
0                       2500     1993   Nonprofit Organization   
1                       5000     1968         Company - Public   
2                       2500     1981  Private Practice / Firm   
3                        750     2000         Company - Public   
4                        125     1998        Company - Private   

                                Industry             Sector  Min_Salary  \
0                     Insurance Carriers          Insurance      137000   
1                 Research & Development  Business Services      137000   
2                             Consulting  Business Services      137000   
3  Electrical & Electronic Manufacturing      Manufacturing      137000   
4                Advertising & Marketing  Business Services      137000   

   Max_Salary  Average_Salary  Revenue_in_Million  Python  Excel  AWS  \
0      171000        154000.0         1234.462963       0      0    1   
1      171000        154000.0         1500.000000       0      0    0   
2      171000        154000.0          300.000000       1      1    1   
3      171000        154000.0          300.000000       1      1    1   
4      171000        154000.0         1234.462963       1      1    0   

   Tableau  SQL  Hadoop  Company_age Job_state  
0        0    0       0           30        NY  
1        0    1       1           55        VA  
2        0    0       0           42        MA  
3        0    1       0           23        MA  
4        0    1       0           25        NY

In [49]:
data.to_csv("C:\\Users\\gaber\\Downloads\\Uncleaned_DS_jobs\\Cleaned_DS_jobs.csv")